packages/mui-base/src/FormControl/FormControl.tsx

```diff
@@ -126,6 +126,15 @@ const FormControl = React.forwardRef(function FormControl<
         setValue(event.target.value);
         onChange?.(event);
       },
+      // TODO: make it like this? to work with SelectUnstyled as well
```

Comment:
Yup, I think this could work if we have all `onChange` handlers with the new value as the second parameter.